### ISSUE TITLE
Menu closes after rotate on Honeycomb

### ISSUE THREAD

**Issue by Nukehead** [JakeWharton/ActionBarSherlock#547]
I have a simple SherlockFragmentActivity with a menu with a single entry. When the menu is opened and the device is rotated, the menu will stay open on GB and ICS but not on 3.2. ABS version used was 4.1.0.

Logcat contains an exception on 3.2 when rotating with and without the menu being opened:

``` text
07-11 11:53:10.962: W/Resources(5639): Converting to string: TypedValue{t=0x1/d=0x103011c a=-1 r=0x103011c}
07-11 11:53:10.971: W/MenuInflater(5639): Cannot instantiate class: @16974108
07-11 11:53:10.971: W/MenuInflater(5639): java.lang.ClassNotFoundException: @16974108 in loader dalvik.system.PathClassLoader[/data/app/com.example.menurotator-2.apk]
07-11 11:53:10.971: W/MenuInflater(5639):   at dalvik.system.PathClassLoader.findClass(PathClassLoader.java:251)
07-11 11:53:10.971: W/MenuInflater(5639):   at java.lang.ClassLoader.loadClass(ClassLoader.java:540)
07-11 11:53:10.971: W/MenuInflater(5639):   at java.lang.ClassLoader.loadClass(ClassLoader.java:500)
07-11 11:53:10.971: W/MenuInflater(5639):   at com.actionbarsherlock.view.MenuInflater$MenuState.newInstance(MenuInflater.java:463)
07-11 11:53:10.971: W/MenuInflater(5639):   at com.actionbarsherlock.view.MenuInflater$MenuState.readItem(MenuInflater.java:366)
07-11 11:53:10.971: W/MenuInflater(5639):   at com.actionbarsherlock.view.MenuInflater.parseMenu(MenuInflater.java:147)
07-11 11:53:10.971: W/MenuInflater(5639):   at com.actionbarsherlock.view.MenuInflater.inflate(MenuInflater.java:97)
07-11 11:53:10.971: W/MenuInflater(5639):   at com.example.menurotator.MainActivity.onCreateOptionsMenu(MainActivity.java:23)
07-11 11:53:10.971: W/MenuInflater(5639):   at android.support.v4.app._ActionBarSherlockTrojanHorse.onCreatePanelMenu(_ActionBarSherlockTrojanHorse.java:45)
07-11 11:53:10.971: W/MenuInflater(5639):   at com.actionbarsherlock.ActionBarSherlock.callbackCreateOptionsMenu(ActionBarSherlock.java:556)
07-11 11:53:10.971: W/MenuInflater(5639):   at com.actionbarsherlock.internal.ActionBarSherlockCompat.preparePanel(ActionBarSherlockCompat.java:483)
07-11 11:53:10.971: W/MenuInflater(5639):   at com.actionbarsherlock.internal.ActionBarSherlockCompat.dispatchInvalidateOptionsMenu(ActionBarSherlockCompat.java:273)
07-11 11:53:10.971: W/MenuInflater(5639):   at com.actionbarsherlock.internal.ActionBarSherlockCompat$1.run(ActionBarSherlockCompat.java:988)
07-11 11:53:10.971: W/MenuInflater(5639):   at android.os.Handler.handleCallback(Handler.java:587)
07-11 11:53:10.971: W/MenuInflater(5639):   at android.os.Handler.dispatchMessage(Handler.java:92)
07-11 11:53:10.971: W/MenuInflater(5639):   at android.os.Looper.loop(Looper.java:132)
07-11 11:53:10.971: W/MenuInflater(5639):   at android.app.ActivityThread.main(ActivityThread.java:4123)
07-11 11:53:10.971: W/MenuInflater(5639):   at java.lang.reflect.Method.invokeNative(Native Method)
07-11 11:53:10.971: W/MenuInflater(5639):   at java.lang.reflect.Method.invoke(Method.java:491)
07-11 11:53:10.971: W/MenuInflater(5639):   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:841)
07-11 11:53:10.971: W/MenuInflater(5639):   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:599)
07-11 11:53:10.971: W/MenuInflater(5639):   at dalvik.system.NativeStart.main(Native Method)
```

Sample project is available if needed.

**Comment by app2go** [comment]
Same here (Android 3.2.1 HTC Flyer and 3.2 Samsung Galaxu Tab 10.1). Happens when the fragments are resumed (screen rotated -> activity restarts, fragments are reattached or activity is resumed (other activity started and terminated)).

W/Resources( 4869): Converting to string: TypedValue{t=0x1/d=0x103011c a=-1 r=0x103011c}
W/MenuInflater( 4869): Cannot instantiate class: @16974108
W/MenuInflater( 4869): java.lang.ClassNotFoundException: @16974108 in loader dalvik.system.PathClassLoader[/data/app/com.onegravity.k10-1.apk]
W/MenuInflater( 4869):  at dalvik.system.PathClassLoader.findClass(PathClassLoader.java:251)
W/MenuInflater( 4869):  at java.lang.ClassLoader.loadClass(ClassLoader.java:540)
W/MenuInflater( 4869):  at java.lang.ClassLoader.loadClass(ClassLoader.java:500)
W/MenuInflater( 4869):  at com.actionbarsherlock.view.MenuInflater$MenuState.newInstance(MenuInflater.java:463)
W/MenuInflater( 4869):  at com.actionbarsherlock.view.MenuInflater$MenuState.readItem(MenuInflater.java:366)
W/MenuInflater( 4869):  at com.actionbarsherlock.view.MenuInflater.parseMenu(MenuInflater.java:147)
W/MenuInflater( 4869):  at com.actionbarsherlock.view.MenuInflater.inflate(MenuInflater.java:97)
W/MenuInflater( 4869):  at com.onegravity.app.activity.MyFragment.onCreateOptionsMenu(MyFragment.java:645)
W/MenuInflater( 4869):  at android.support.v4.app._ActionBarSherlockTrojanHorse.onCreatePanelMenu(_ActionBarSherlockTrojanHorse.java:56)
W/MenuInflater( 4869):  at com.actionbarsherlock.ActionBarSherlock.callbackCreateOptionsMenu(ActionBarSherlock.java:556)
W/MenuInflater( 4869):  at com.actionbarsherlock.internal.ActionBarSherlockCompat.preparePanel(ActionBarSherlockCompat.java:483)
W/MenuInflater( 4869):  at com.actionbarsherlock.internal.ActionBarSherlockCompat.dispatchInvalidateOptionsMenu(ActionBarSherlockCompat.java:273)
W/MenuInflater( 4869):  at com.actionbarsherlock.internal.ActionBarSherlockCompat$1.run(ActionBarSherlockCompat.java:988)
W/MenuInflater( 4869):  at android.os.Handler.handleCallback(Handler.java:587)
W/MenuInflater( 4869):  at android.os.Handler.dispatchMessage(Handler.java:92)
W/MenuInflater( 4869):  at android.os.Looper.loop(Looper.java:152)
W/MenuInflater( 4869):  at android.app.ActivityThread.main(ActivityThread.java:4636)
W/MenuInflater( 4869):  at java.lang.reflect.Method.invokeNative(Native Method)
W/MenuInflater( 4869):  at java.lang.reflect.Method.invoke(Method.java:491)
W/MenuInflater( 4869):  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:841)
W/MenuInflater( 4869):  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:599)
W/MenuInflater( 4869):  at dalvik.system.NativeStart.main(Native Method)

**Comment by app2go** [comment]
Same issue as #540

**Comment by JakeWharton** [comment]
Fixed with #540.
